prow/flagutil/core_config.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// CoreConfigOptions holds options for interacting with Configs.
+type CoreConfigOptions struct {
+	ConfigPath    string
+	JobConfigPath string
+}
+
+// AddFlags injects config options into the given FlagSet.
+func (o *CoreConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.ConfigPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	fs.StringVar(&o.JobConfigPath, "job-config-path", "", "Path to prow job configs.")
+}
+
+// Validate validates config options.
+func (o *CoreConfigOptions) Validate(dryRun bool) error {
+	return nil
+}
+
+// Agent returns a started config agent.
+func (o *CoreConfigOptions) Agent() (agent *config.Agent, err error) {
+	agent = &config.Agent{}
+	config, err := config.Load(o.ConfigPath, o.jobConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	agent.Set(config)
```

Comment:
I don't think we need to do the `Load` and `Set` -- the first call to `Start` will handle that for us

Reply:
(and `Load` calls `Set` for plugin config anyway)

Reply:
Can you be a little more specific here, please? I think we need to just do a `config.Load()`, right?